src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -1898,6 +1898,7 @@ trait Contexts { self: Analyzer =>
 
     /** If requireExplicit is true, wildcard imports are not considered. */
     def importedSelectedSymbol(name: Name, requireExplicit: Boolean): (ImportSelector, Symbol) = {
+      if (isRootImport && qual.symbol.hasPackageFlag) openPackageModule(qual.symbol, force = true)
```

Comment:
Why do we need a special case for root imports, do you know?

I tried two files
```
package a {
  package object p {
    type Sr = java.io.Serializable
  }
}
```

```
import a.p._
package b {
  package object q extends Sr
}
```

When compiling the second, `openPackageModule` for `a.b` is called in `PackageLoader.doComplete`. Ideas why this doesn't happen for root imports?

Reply:
Root imports don't benefit from namer. The nicer place to set this up is when creating the root context: intuitively, do the work when completing the lazy type. I don't see yet how to do that for `Predef`, but I'll continue fiddling.

Reply:
So the `PackageLoader` for `scala` is forced very early (before namer) due to things in Definitions (`ObjectClass.initialize` triggers a bunch of stuff through the classfile parser).

This means that opening the package object is deferred for `scala`.

The lookup for `Serializable` when compiling `package object XX extends Serializable` happens in phase `packageobjects`, which does

```
        openPackageObjectsTraverser(unit.body)     // triggers lookup
        deferredOpen.foreach(openPackageModule(_)) // too late
```

This order is deliberate though (https://github.com/scala/scala/pull/9661/files).

The "root imports" focus of this PR is probably a red herring? I can compile `test/files/pos/t12740b` below using 2.13.12 with `-Yimports` without problem.

Mabye we just need to force the `scala` package early? Here's the list of packages that are on initialization (even before parser):
```
<root>
java
java.lang
jdk
jdk.internal
jdk.internal.vm
jdk.internal.vm.annotation
> scala
scala.annotation
scala.runtime
> scala.collection
> scala.collection.immutable
java.io
java.lang.constant
```

Then two more during namer

```
> scala.reflect
scala.annotation.meta
```

I marked the ones that have a non-empty package object.

Reply:
So I think my attempt could partially subvert the fix added in https://github.com/scala/scala/pull/9661. According to the PR description, if we had

`package object scala extends SP`

and we'd re-compile the source file where `SP` is defined but not the pacakge object, we could run into a spurious "symbol not found" error.

But the few package objects mentioned above are are all without parent, and anyway they are all in our source code, so my commit is not reverting #9661 for user code.

Reply:
I haven't had a chance to revisit this and cope with the "red herring" remark. To me, this was about a uniform mechanism for `-Yimports`.